akka-http-caching/src/main/scala/akka/http/scaladsl/server/directives/CachingDirectives.scala

```diff
@@ -49,13 +49,13 @@ trait CachingDirectives {
     // Do directive processing asynchronously to avoid locking the cache accidentally (#4092)
     // This will be slightly slower, but the rational here is that caching is used for slower kind of processing
     // anyway so the performance hit should be acceptable.
-    CachingDirectives.asyncRoute &
```

Comment:
Might have worked reversing the two directives. This seems simpler. Interesting that it only failed for Akka 2.6 and still not at all times... I reproduced it using Akka 2.6.18 and using `taskset -c 0` to limit CPU resources.